Divine/CLI/CommandLineArguments.cs

```diff
@@ -0,0 +1,327 @@
+        {
+            ResourceFormat resourceFormat;
+
+            switch (optionResourceFormat)
+            {
+                case "lsb":
+                    resourceFormat = ResourceFormat.LSB;
+                    break;
+                case "lsf":
+                    resourceFormat = ResourceFormat.LSF;
+                    break;
+                case "lsj":
+                    resourceFormat = ResourceFormat.LSJ;
+                    break;
+                case "lsx":
+                    resourceFormat = ResourceFormat.LSX;
+                    break;
+                default:
+                    resourceFormat = ResourceFormat.LSX;
+                    break;
+            }
+
+            return resourceFormat;
+        }
+
+        public static PackageVersion GetPackageVersion(string versionOption)
+        {
+            PackageVersion version;
+
+            switch (versionOption)
+            {
+                case "v7":
+                    version = Enums.PackageVersion.V7;
+                    break;
+
+                case "v9":
+                    version = Enums.PackageVersion.V9;
+                    break;
+
+                case "v10":
+                    version = Enums.PackageVersion.V10;
+                    break;
+
+                case "v13":
+                    version = Enums.PackageVersion.V13;
+                    break;
+
+                default:
+                    version = Enums.PackageVersion.V13;
+                    break;
+            }
+
+            return version;
+        }
+
+        public static Dictionary<string, object> GetCompressionOptions(string compressionOption, PackageVersion version)
+        {
+            CompressionMethod compression;
+            bool fastCompression;
+
+            switch (compressionOption)
+            {
+                case "zlibfast":
+                    compression = LSLib.LS.Enums.CompressionMethod.Zlib;
+                    fastCompression = true;
+                    break;
+
+                case "zlib":
+                    compression = LSLib.LS.Enums.CompressionMethod.Zlib;
+                    fastCompression = false;
+                    break;
+
+                case "lz4":
+                    compression = LSLib.LS.Enums.CompressionMethod.LZ4;
+                    fastCompression = true;
+                    break;
+
+                case "lz4hc":
+                    compression = LSLib.LS.Enums.CompressionMethod.LZ4;
+                    fastCompression = false;
+                    break;
+
+                default:
+                    compression = LSLib.LS.Enums.CompressionMethod.None;
+                    fastCompression = false;
+                    break;
+            }
+
+            // fallback to zlib, if the package version doesn't support lz4
+            if (compression == LSLib.LS.Enums.CompressionMethod.LZ4 && version <= (PackageVersion) 9)

```

Comment:
wouldn't `version <= PackageVersion.V9` be better?

Reply:
Yessir.